Sources/Commands/SwiftBuildTool.swift

```diff
@@ -157,6 +162,29 @@ public struct SwiftBuildTool {
             }
         
             switch mode {
+            case .repl(let toolchain):
+                let (rootPackage, externalPackages) = try fetch(opts.path.root)
+                let (modules, externalModules, _) = try transmute(rootPackage, externalPackages: externalPackages)
+
+                /// TODO: Need to create DSOs for C executables as well.
+                let products: [Product] = modules.flatMap{ $0 as? SwiftModule }
+                                                 .map{ Product(name: $0.name, type: .Library(.Dynamic), modules: [$0]) }
+
+                let yaml = try describe(opts, .debug, modules, Set(externalModules), products, toolchain: toolchain)
+                try build(YAMLPath: yaml, target: opts.buildTests ? "test" : nil)
+
+                let libdir = "\(opts.path.build)/debug"
+                var args = ["-L", libdir, "-I", libdir]
+                args += toolchain.platformArgs
+                args += modules.map{ return "-l\($0.name)" }
+
+                let finalCommand = [toolchain.lldb, "--repl=" + args.joined(separator: " ") ]
+
+                /// Get the $TERM environment variable.
+                /// This is needed by lldb to start the REPL.
+                let term = getenv("TERM") ?? "dumb"
+                try system(finalCommand, environment: ["TERM": term])
```

Comment:
It probably makes sense to launch `swift` inheriting the full environment in this mode, not specific things we handpick.